IdentityServer3.Contrib.RedisStore/Stores/BaseTokenStore.cs

```diff
@@ -111,6 +111,14 @@ protected T ConvertFromJson(string json)
             {
                 await this.database.StringSetAsync(tokenKey, json, expiresIn).ConfigureAwait(false);
             }
+            
+            //allways cleanup any previous expired keys
```

Comment:
this cleanup is irrelevant in the case of access token issued for Client Credential flow, it should be inside the `if (!string.IsNullOrEmpty(token.SubjectId))` branch.